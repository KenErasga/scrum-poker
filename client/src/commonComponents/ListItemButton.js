import React from 'react';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';

const ListItemButton = ({ description, onClick, Icon }) => {
    return (
        <ListItem button>
            <ListItemIcon>
                <Icon />
            </ListItemIcon>
            <ListItemText primary={description} onClick={onClick} />
        </ListItem>
    );
};

export default ListItemButton;





