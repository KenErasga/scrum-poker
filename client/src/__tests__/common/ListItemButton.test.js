import React from 'react';
import renderer from 'react-test-renderer';
import ListItemButton from '../../common/ListItemButton';

describe('ListItemButton', () => {
    it('ListItemButton renders correctly', () => {
    
    const tree = renderer
        .create(<ListItemButton description='test' onClick={(e) => e} Icon='test' />)
        .toJSON();
    expect(tree).toMatchSnapshot();
    });
});
