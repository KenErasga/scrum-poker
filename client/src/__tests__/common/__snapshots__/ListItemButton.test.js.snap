// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemButton ListItemButton renders correctly 1`] = `
<div
  aria-disabled={false}
  className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="MuiListItemIcon-root"
  >
    <test />
  </div>
  <div
    className="MuiListItemText-root"
    onClick={[Function]}
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
    >
      test
    </span>
  </div>
</div>
`;
